doc/source/reference/routines.io.rst

```diff
@@ -85,7 +86,6 @@ Data sources
 Binary Format Description
 -------------------------
 .. autosummary::
-   :template: autosummary/minimal_module.rst
```

Comment:
why this change? Is there something wrong with the template?

Reply:
It was based on [this comment](https://github.com/numpy/numpy/pull/13414#issuecomment-492908468) from the review of #13414. If no longer necessary I'm happy to revert.

Reply:
Thanks, that overly-terse comment brought me back to [this one](https://github.com/numpy/numpy/pull/13414#issuecomment-487290004) which pointed out that `open_memmap` is [not actually documented](https://numpy.org/devdocs/search.html?q=open_memmap&check_keywords=yes&area=default), but after removing that line in this PR [it is](https://14195-908607-gh.circle-artifacts.com/0/doc/build/html/reference/generated/numpy.memmap.html) (scroll down to the See Also section). I think this is the only reference to that template, so maybe it can be removed. 

On the other hand, the [index page for memmap](https://14195-908607-gh.circle-artifacts.com/0/doc/build/html/reference/routines.io.html#memory-mapping-files) now seems weird: why does the function appear as `lib.format.open_memmap` and not `open_memmap`, and where are the rest of the functions from `lib.format.py` like `read_array`, `write_array` and the docstring for that file?

Reply:
> but after removing that line in this PR it is (scroll down to the See Also section). 

I think this is rather due to the fact that I explicitly added the function `lib.format.open_memmap` to the `routines.io` autosummary section on memmaps (see the other changed line in this PR).

> and where are the rest of the functions from lib.format.py like read_array, write_array and the docstring for that file?

I'm not sure - maybe the functions in `lib.format` are not being picked up by autosummary because the `lib.format.rst` doc is itself auto-generated?

Reply:
I tried several approaches but wasn't able to get the `autosummary` to include the docstrings for the `lib.format` functions in the generated docs. I wonder if they were ever included as it turns out `descr_to_dtype` had an rST formatting error in it which raised a sphinx warning on build.

In any case, eb6f013 presents an alternative which explicitly includes the `lib.format` functions in an autosummary in `routines.io` to ensure the `.rst` stubs for the functions are generated properly.